crates/router/src/db/mandate.rs

```diff
@@ -108,46 +112,150 @@ impl MandateInterface for Store {
+            if let Some(created_time) = mandate_constraints.created_time {
+                checker &= mandate.created_at == created_time;
+            }
+            if let Some(created_time_lt) = mandate_constraints.created_time_lt {
+                checker &= mandate.created_at < created_time_lt;
+            }
+            if let Some(created_time_gt) = mandate_constraints.created_time_gt {
+                checker &= mandate.created_at > created_time_gt;
+            }
+            if let Some(created_time_lte) = mandate_constraints.created_time_lte {
+                checker &= mandate.created_at <= created_time_lte;
+            }
+            if let Some(created_time_gte) = mandate_constraints.created_time_gte {
+                checker &= mandate.created_at >= created_time_gte;
+            }
+            if let Some(connector) = &mandate_constraints.connector {
+                checker &= mandate.connector == *connector;
+            }
+            if let Some(mandate_status) = mandate_constraints.mandate_status {
+                let storage_mandate_status: storage_models::enums::MandateStatus =
+                    mandate_status.foreign_into();
+                checker &= mandate.mandate_status == storage_mandate_status;
+            }
+            checker
+        });
+
+        let mandates: Vec<storage::Mandate> = if let Some(limit) = mandate_constraints.limit {
+            #[allow(clippy::as_conversions)]
+            mandates_iter.take(limit as usize).cloned().collect()
+        } else {
+            mandates_iter.cloned().collect()
+        };
+        Ok(mandates)
     }
 
     async fn insert_mandate(
         &self,
-        _mandate: storage::MandateNew,
+        mandate_new: storage::MandateNew,
     ) -> CustomResult<storage::Mandate, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let mut mandates = self.mandates.lock().await;
+        let mandate = storage::Mandate {
+            #[allow(clippy::as_conversions)]
+            id: mandates.len() as i32,
+            mandate_id: mandate_new.mandate_id.clone(),
+            customer_id: mandate_new.customer_id,
+            merchant_id: mandate_new.merchant_id,
+            payment_method_id: mandate_new.payment_method_id,
+            mandate_status: mandate_new.mandate_status,
+            mandate_type: mandate_new.mandate_type,
+            customer_accepted_at: mandate_new.customer_accepted_at,
+            customer_ip_address: mandate_new.customer_ip_address,
+            customer_user_agent: mandate_new.customer_user_agent,
+            network_transaction_id: mandate_new.network_transaction_id,
+            previous_attempt_id: mandate_new.previous_attempt_id,
+            created_at: mandate_new.created_at.unwrap_or_else(|| {
+                let now = OffsetDateTime::now_utc();
+                PrimitiveDateTime::new(now.date(), now.time())
+            }),
```

Comment:
You can use the `common_utils::date_time::now()` funtion instead.

```suggestion
            created_at: mandate_new.created_at.unwrap_or_else(common_utils::date_time::now),
```

You might have to run the formatter though.